csrc/executor.cpp

```diff
@@ -1635,6 +1642,19 @@ std::vector<at::Tensor> FusionExecutor::runFusion(
   }
   args.push(outputs);
 
+  for (const auto i : c10::irange(outputs.size())) {
+    auto output = kernel()->outputs()[i];
+    if (std::any_of(
+            kernel()->inputs().begin(),
+            kernel()->inputs().end(),
+            [&](const auto& in) { return in == output; })) {
+      // Skip trivially forwarded outputs because they are just placeholders
```

Comment:
I don't see how this is related to `finalize`. I think the problem is, we need the `executor_utils::bindInputForExprEvaluation` for non-forwarded outputs, but not trivially forwarded outputs. I don't see how `finalize` can help on that.

Reply:
Oh I just meant that we could detect whether each output is a simple forwarded input inside `finalize` to speed up `runFusion`.

Reply:
I would generally not consider performance optimization unless the added complexity is negligible. For this, we would likely need to add another field to the `Kernel` marking which is trivially forwarded. This sounds too much of a complexity to have, it also make our code less modular because `Kernel` contains a cache for executor.

I think for general development, we should focus on functionality correctness and code simplicity, because anything we write today might subject to change in the future, and the simpler and modular we can make these things, the easier it would be for refactoring. We should only optimize performance when it becomes an issue, and during performance optimization, we should identify what is the slowest part and try to improve it, rather than implementing all possible micro-optimizations at the price of code complexity.

Reply:
Makes sense. Sounds good to me.